### DISCUSSION TITLE
Elsa 3 - custom activity property editor

### DISCUSSION THREAD

**Discussion by lemac1** [elsa-workflows/elsa-core#4221] [Q&A]
Hello,

I would like to write in angular custom editor of property of custom activity in elsa3.
As base i'm using angular-workspace sample from git repository (v3-RC1).

I have managed to write custom input driver, the problem is  while rendering. 
What should return this method? 
renderInput(context: ActivityInputContext)

While returning string - it is printed as string, Html tags in string are also printed as string. any elements, components etc are printed as undefined. I need to put here custom component with logic.

**Comment by sfmskywalker** [comment]
Hello,

Thank you for reaching out and experimenting with custom input drivers for Elsa 3 using Angular.

1. The `renderInput` method of the `ActivityDriver` interface is designed to return a JSX object, which is specific to StencilJS. 
2. Given the current design, it unfortunately means you cannot write custom drivers using Angular directly.
3. We could certainly consider modifying this behavior so that if a string is returned, it's handled appropriately, allowing for more flexibility.
4. It's worth noting that the StencilJS implementation of the workflow designer is on track to be replaced with a Blazor implementation of Elsa Studio in future releases.
5. With this change, Blazor components will be provided as custom elements. While this allows for integration with Angular, it means you'd need to write custom input handlers using C# instead of Angular.

I understand this might not be the ideal solution you were hoping for. We are continuously working on improving Elsa and your feedback is valuable for our roadmap. Please let us know if you have any further questions or suggestions.
